### ISSUE TITLE
With selective copying, if you have several instances of the same module, the order of the modules is sorted by access in history and not by hierarchical order

### ISSUE THREAD

**Issue by s7habo** [darktable-org/darktable#16372]
### Describe the bug

With selective copying, if you have several instances of the same module, the order of the modules is sorted by access in history and not by hierarchical or custom order. 

### Steps to reproduce

1. Make the changes in the first instance of a module. In this case color balance rgb:

![grafik](https://github.com/darktable-org/darktable/assets/38165484/2e4e2254-216e-472f-9d77-ef69a5d02ad0)

2. Add a **new instance** and make the change there too. Rename it. Here it is called "blue to cyan":

![grafik](https://github.com/darktable-org/darktable/assets/38165484/0aa151f8-c097-4cec-a26d-81fdd3329bd9)

3. As expected, "blue to cyan" is **hierarchically higher** than the first instance. We want to **keep this order** when inserting it into the new photo.

![grafik](https://github.com/darktable-org/darktable/assets/38165484/6da128b1-b1c5-4bfc-b3a9-9fe2b6d71403)

4. We now go back to the **first instance** and make some  additional changes there:

![grafik](https://github.com/darktable-org/darktable/assets/38165484/3984d80f-4f7e-421e-ab94-b04d49d5d698)

5, We go to the lighttable and select the option "selective copy". Notice how the first instance is now at the top of the list:

![grafik](https://github.com/darktable-org/darktable/assets/38165484/d3436db8-ff70-48e1-b167-1eebbe5ad8c5)

6. If we now paste this into the new photo, this order will be **copied from the history into the hierarchy**, which we do not want.  Notice how the "blue to cyan" instance is now **lower in the hierarchy** than the first instance, which is now also numbered "1":

![grafik](https://github.com/darktable-org/darktable/assets/38165484/6d7de759-4585-4a45-8d69-c8d21dfff3f0)

### Expected behavior

The order of the instances should be structured according to the hierarchy and not according to the access time.

### Where did you obtain darktable from?

self compiled

### darktable version

4.7.0+629~gc07dd92d7

### What OS are you using?

Linux

### What is the version of your OS?

Ubuntu Studio 22.04

### Describe your system?

Operating System: Ubuntu Studio 22.04
KDE Plasma Version: 5.24.7
KDE Frameworks Version: 5.92.0
Qt Version: 5.15.3
Kernel Version: 5.19.0-1028-lowlatency (64-bit)
Graphics Platform: X11
Processors: 8 × Intel® Core™ i7-7700K CPU @ 4.20GHz
Memory: 31.3 GiB of RAM
Graphics Processor: NVIDIA GeForce GTX 1070/PCIe/SSE2

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

NVIDIA GeForce GTX 1070/PCIe/SSE2, 8192 MB, 535.154.05

**Comment by s7habo** [comment]
>  I hat those bugs so I kill them.

Yes, I understand.  Those nasty little critters that spoil one's mood. I was working on a new episode today and then... 

> One less, is that the end of this?

I'll be sure to light it up for you when I have to lift the carpet again and one crosses my path. They won't be crawling for long. :)

**Comment by wpferguson** [comment]
@TurboGit sorry to rain on your parade, but I found another history copy/paste "feature".  I just tested #16373 to see if that fixed it, but sadly not.  I'll raise another issue.
